src/main/javascript/modules/area/asgarnia/dark-wizards-tower.ts

```diff
@@ -0,0 +1,161 @@
+    _entity.setCoords(ctx.player, _coords(0, 45, 52, 18, 40));
+});
+
+_events.bindEventListener(EventType.OPLOC1, 24681, (ctx) => {//Drawers in Witch's House
+    _map.addLoc(24682, _map.getCoords(ctx.location), getLocShape(ctx.location), getLocRotation(ctx.location));
+});
+
+_events.bindEventListener(EventType.OPLOC2, 24682, (ctx) => {//Drawers in Witch's House
+    sendMessage(ctx.player, "The drawers are empty.");
+});
+
+_events.bindEventListener(EventType.OPLOC3, 24682, (ctx) => {//Drawers in Witch's House
+    _map.addLoc(24681, _map.getCoords(ctx.location), getLocShape(ctx.location), getLocRotation(ctx.location));
+	sendMessage(ctx.player, "You shut the drawers.");
+});
+
+_events.bindEventListener(EventType.OPLOC1, 24718, (ctx) => {//Ladder in Witch's House
+    runAnim(ctx.player, 828, function () {
+		_entity.setCoords(ctx.player, _coords(2774, 9759, 0));
+	});
+});
+
+_events.bindEventListener(EventType.OPLOC1, 24882, (ctx) => {//Gramophone in Witch's House
+    sendMessage(ctx.player, "The gramophone doesn't have a record on it.");
+});
+
+_events.bindEventListener(EventType.OPLOC1, 24842, (ctx) => {//Manhole (Quest Witch's House)
+	if (mapMembers()){
+	    sendMessage(ctx.player, "The cover won't budge. It seems to be locked.");
+    } else {
+        sendMessage(ctx.player, "You need to be on a member's world to use this feature.");
+    }
+});
+
+_events.bindEventListener(EventType.OPLOC1, 2867, async (ctx) => {//Potted plant (Quest Witch's House)
+	if (mapMembers()){
+	    if (hasItem(ctx.player, 2409)) {
+	        sendMessage(ctx.player, "You don't find anything interesting.");
+	    } else {
+	        giveItem(ctx.player, 2409, 1);
+	         await mesbox(ctx.player, "You find a key hidden under the flower pot.");
+	    }
+    } else {
+        sendMessage(ctx.player, "You need to be on a member's world to use this feature.");
+    }
+});
+
+_events.bindEventListener(EventType.OPLOC1, 2861, async (ctx) => {//Door (Quest Witch's House)
+	if (mapMembers()){
+	    if (hasItem(ctx.player, 2409)) {
+	        if(hasStarted(ctx.player, 7)) {
+		        sendMessage(ctx.player, "todo add doors that work");
+		    } else {
+	            await chatplayer(ctx.player, "It would be rude to break into this house.");
+		    }
+	    } else {
+	        sendMessage(ctx.player, "This door is locked.");
+	    }
+    } else {
+        sendMessage(ctx.player, "You need to be on a member's world to use this feature.");
+    }
+});
+
+_events.bindEventListener(EventType.OPLOC1, 2862, async (ctx) => {//door to garden(Quest Witch's House)
+	if (varp(ctx.player, 2276) == 3) {
+		sendMessage(ctx.player, "todo add door ");
+    } else {
+	    await mesbox(ctx.player, "This door is locked.");
+        await chatplayer(ctx.player, "Strange... I can't see any kind of lock or handle to open<br> this door...");
+	}
+});
+
+_events.bindEventListener(EventType.OPLOCU, 2861, async (ctx) => {//door (Quest Witch's House)
+	switch (ctx.objId) {
```

Comment:
This should use `ctx.useObjId` - there's no `objId` property on the `OPLOCU` event. You might have to add `useObjId` to `EventContext`